amethyst_core/src/transform/systems.rs

```diff
@@ -150,20 +150,15 @@ mod tests {
     #[test]
     fn transform_matrix() {
         let mut transform = Transform::default();
-        transform.translation = Vector3::new(5.0, 2.0, -0.5);
-        transform.rotation = Quaternion::new(1.0, 0.0, 0.0, 0.0);
-        transform.scale = Vector3::new(2.0, 2.0, 2.0);
-
-        let decomposed = Decomposed {
-            rot: transform.rotation,
-            disp: transform.translation,
-            scale: 2.0,
-        };
+        transform.set_xyz(5.0, 2.0, -0.5);
+        transform.set_rotation(Unit::new_normalize(Quaternion::new(1.0, 0.0, 0.0, 0.0)));
```

Comment:
Isn't this quaternion different from the previous one? I saw previously in this pr that at some point the last value was swapped with the first...

Reply:
They are all supposed to be initialized this way but read from the internal array as [x, y, z, w]